libsolv-sys/build.rs

```diff
@@ -36,34 +36,36 @@ fn main() {
 
         // Whitelist libsolv's functions, types, and variables,
         // otherwise bindgen will bind all of libc
-        .whitelisted_type("Solver")
-        .whitelisted_type("Chksum")
-        .whitelisted_type("solv.*")
-        .whitelisted_function("pool.*")
-        .whitelisted_function("stringpool.*")
-        .whitelisted_function("transaction.*")
-        .whitelisted_function("solv.*")
-        .whitelisted_function("selection.*")
-        .whitelisted_function("repopagestore.*")
-        .whitelisted_function("repo.*")
-        .whitelisted_function("queue.*")
-        .whitelisted_function("policy.*")
-        .whitelisted_function("find.*")
-        .whitelisted_function("dirpool.*")
-        .whitelisted_function("datamatcher.*")
-        .whitelisted_function("dataiterator.*")
-        .whitelisted_function("map.*")
-        .whitelisted_var("DI.*")
-        .whitelisted_var("SOLV.*")
-        .whitelisted_var("REPO.*")
-        .whitelisted_var("SEARCH.*")
-        .whitelisted_var("EVRCMP.*")
+        .whitelist_type("Solver")
+        .whitelist_type("Chksum")
+        .whitelist_type("solv.*")
+        .whitelist_function("pool.*")
+        .whitelist_function("stringpool.*")
+        .whitelist_function("transaction.*")
+        .whitelist_function("solv.*")
+        .whitelist_function("selection.*")
+        .whitelist_function("repopagestore.*")
+        .whitelist_function("repo.*")
+        .whitelist_function("queue.*")
+        .whitelist_function("policy.*")
+        .whitelist_function("find.*")
+        .whitelist_function("dirpool.*")
+        .whitelist_function("datamatcher.*")
+        .whitelist_function("dataiterator.*")
+        .whitelist_function("map.*")
+        .whitelist_var("DI.*")
+        .whitelist_var("SOLV.*")
+        .whitelist_var("REPO.*")
+        .whitelist_var("SEARCH.*")
+        .whitelist_var("EVRCMP.*")
 
         // Hide FILE from bindgen's output
         // Otherwise we get the OS's private file implementation
-        .hide_type("FILE")
+        .blacklist_type("FILE")
         .raw_line("use libc::FILE;")
 
+        .rustified_enum("solv_knownid")
```

Comment:
I really have no idea if this is an appropriate thing to do, but it solved the compile error I was seeing, and I don't understand any implications it might have outside of that.

Reply:
I'm not sure either, but we'll figure it out! Thanks so much for getting it compiling again!

Reply:
Whoops!  There was one compile error left that I hadn't pushed yet, I was initially working on the "rewiring" branch because I hadn't noticed this one.

It looks like you're already on that, though

Reply:
@ambaxter https://github.com/ambaxter/libsolv-rs/pull/3